WooCommerce/Classes/POS/Presentation/PointOfSaleDashboardView.swift

```diff
@@ -34,8 +34,10 @@ struct PointOfSaleDashboardView: View {
         }
         .sheet(isPresented: $viewModel.showsCardReaderSheet, content: {
             switch viewModel.cardPresentPaymentEvent {
-            case .showAlert(let alertViewModel):
-                CardPresentPaymentAlert(alertViewModel: alertViewModel)
+            case .showAlert(let alertDetails):
+                CardPresentPaymentAlert(
```

Comment:
If it's easier to reuse the existing payment alert UI, certain cases of `CardPresentPaymentAlertDetails` could be converted to their corresponding `CardPresentPaymentsModalViewModel` implementation. Maybe some mapper from `CardPresentPaymentAlertDetails` to some presentation type would be helpful, taking connectReader/collectPayment use case into consideration to determine the UI to be shown.

Reply:
Definitely possible, and maybe for most connection alerts, which are due to keep their design, we should do that for now. Those where we need to specialise because of incorrect action behaviour, we can handle separately.

For payment alerts, I think we should specialise.

Reply:
@jaclync I decided against this... the benefit of the specific view models is too great, as we can make them much more declarative and avoid all the optionals present in the legacy view model. It doesn't take that long to convert the view models and make views for them.

Reply:
Yea, the missing action behavior like dismissal is tricky and it makes sense to just have a custom view model 👍